lib/watir-webdriver/checkers.rb

```diff
@@ -0,0 +1,88 @@
+module Watir
+  class Checkers
+
+    def initialize(browser)
+      @browser = browser
+      @error_checkers = []
+    end
+
+    #
+    # Adds new checker.
+    #
+    # Checkers are generally used to ensure application under test does not encounter
+    # any error. They are automatically executed after following events:
+    #   1. Open URL.
+    #   2. Refresh page.
+    #   3. Click, double-click or right-click on element.
+    #
+    # @example
+    #   browser.checkers.add do |page|
+    #     page.text.include?("Server Error") and puts "Application exception or 500 error!"
+    #   end
+    #   browser.goto "www.watir.com/404"
+    #   "Application exception or 500 error!"
+    #
+    # @param [#call] checker Object responding to call
+    # @yield Checker block
+    # @yieldparam [Watir::Browser]
+    #
+
+    def add(checker = nil, &block)
```

Comment:
Interface looks weird to me and I also think that currently you can easily do similar things since checker should just respond to `#call`:

```ruby
class MyChecker
  def call(browser)
    # ...
  end
end

browser.add_checker MyChecker.new
```